lib/semantics/semantics.h

```diff
@@ -172,6 +177,7 @@ class SemanticsContext {
 
   bool CheckError(bool);
   ConstructStack constructStack_;
+  std::map<const Symbol *, const parser::CharBlock> activeDoVariables_;
```

Comment:
If the do variable symbols are changed to refs, the key for this map should then be `SymbolRef`.

Reply:
Sounds good.